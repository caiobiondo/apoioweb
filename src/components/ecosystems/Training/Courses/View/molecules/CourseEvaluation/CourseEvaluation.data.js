import gql from 'graphql-tag';
import { Origem } from '../../../../../../../config';

export const CourseEvaluationQuery = gql`
  query CourseEvaluationtQuery(
    $sellerId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    trainingEvaluations(
      sellerId: $sellerId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      id
      description
      order
    }
  }
`;

export const CourseAddEvaluationMutation = gql`
  mutation AddCourseEvaluations(
    $sellerId: Int!
    $courseId: Int!
    $ciclo: Int!
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
    $input: AddCourseEvaluationInput!
  ) {
    addCourseEvaluations(
      sellerId: $sellerId
      courseId: $courseId
      ciclo: $ciclo
      input: $input
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      status
      message
    }
  }
`;

export const CourseEvaluationQueryOptions = {
  options(props) {
    return {
      forceFetch: true,
      sellerId: props.user.codigo,
      ciclo:
        props.user.estrutura.ciclo &&
        props.user.estrutura.ciclo[0] &&
        props.user.estrutura.ciclo[0].numero,
      setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
      gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
      grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
      papelDaConsultora: props.user.cdPapelAtivo,
      canal: props.user.cdCanalCaptacao,
      gerenciaDeVendas:
        props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
      regiao:
        props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
      origem: Origem,
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      evaluations: data.trainingEvaluations,
    };
  },
};
