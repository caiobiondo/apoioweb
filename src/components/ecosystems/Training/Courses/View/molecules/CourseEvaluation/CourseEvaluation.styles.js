import styled from 'styled-components';
import {
  RobotoRegular,
  NaturaBold,
  NaturaRegular,
  fs15,
  fs16,
  fs25,
  fw700,
} from 'styles/typography';
import { gray700 } from 'styles/colors';

export const CourseEvaluationModalTitle = {
  fontFamily: NaturaBold,
  fontSize: fs25,
};

export const CourseEvaluationModalAction = {
  fontFamily: RobotoRegular,
  fontWeight: fw700,
};

export const ContentWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const TitleWrapper = styled.div`
  font-family: ${RobotoRegular};
  font-size: ${fs16};
  color: ${gray700};
`;

export const RatingWrapper = styled.div`
  div {
    display: flex;
  }

  svg {
    height: 15px !important;
    width: 15px !important;
  }
`;

export const WrapperCount = styled.div`
  position: absolute;
  right: 30px;
  top: 30px;
  font-family: ${NaturaRegular};
  font-size: ${fs15};
`;
