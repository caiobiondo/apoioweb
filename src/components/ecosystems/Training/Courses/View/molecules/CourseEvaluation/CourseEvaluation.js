import React, { Component } from 'react';
import { graphql } from 'react-apollo';
import { Dialog, FlatButton, Rating, Loading } from 'natura-ui';
import { FormattedMessage } from 'react-intl';
import { translate } from 'locale';
import { CourseEvaluationtQuery, CourseEvaluationtQueryOptions } from './CourseEvaluation.data';
import {
  CourseEvaluationModalTitle,
  CourseEvaluationModalAction,
  TittleWrapper,
  RatingWrapper,
  RatingStyles,
  ContentWrapper,
  WrapperCount,
} from './CourseEvaluation.styles';

class CourseEvaluation extends Component {
  state = {
    modalOpened: true,
    currentIndex: 0,
  };

  componentWillReceiveProps({ loading, evaluations }) {
    this.notifyLoadFinish(loading, evaluations);
  }

  notifyLoadFinish = (loading, evaluations) => {
    if (!loading && this.props.onLoadFinished) {
      this.props.onLoadFinished(
        this.isEmpty(loading, evaluations),
        this.isLoading(loading, evaluations),
      );
    }
  };

  handleClose = () => {
    if (this.isLastEvaluation()) this.setState({ modalOpened: false });
    if (!this.isLastEvaluation())
      this.setState({
        currentIndex: this.state.currentIndex + 1,
      });
  };

  isLoading = (loading, evaluations) => loading && !evaluations;

  isEmpty = (loading, evaluations) => !loading && (!evaluations || evaluations.length === 0);

  isLastEvaluation = () => this.totalOfEvaluations() - 1 === this.state.currentIndex;

  totalOfEvaluations = () => this.props.evaluations.length;

  defineModalActions = () => {
    const dynamicLabel = this.isLastEvaluation() ? 'courseEvaluationEnd' : 'courseEvaluationNext';
    const dynamicProps = {
      label: <FormattedMessage id={dynamicLabel} />,
      labelStyle: CourseEvaluationModalAction,
      primary: true,
      onClick: this.handleClose,
    };

    return [
      <FlatButton
        label={<FormattedMessage id="cancel" />}
        labelStyle={CourseEvaluationModalAction}
        onClick={this.handleClose}
      />,
      <FlatButton {...dynamicProps} />,
    ];
  };

  render() {
    const { evaluations } = this.props;

    if (!evaluations && this.props.loading) {
      return <Loading background="transparent" />;
    }

    const currentIndex = this.state.currentIndex;
    const counter = `${currentIndex + 1}/${this.totalOfEvaluations()}`;
    const evaluation = evaluations[currentIndex];

    const actions = this.defineModalActions();

    return (
      <Dialog
        key="evaluationkModal"
        title={translate('courseEvaluationTitle')}
        actions={actions}
        modal={false}
        open={this.state.modalOpened}
        titleStyle={CourseEvaluationModalTitle}
        onRequestClose={this.handleClose}
      >
        <ContentWrapper>
          <TittleWrapper>{evaluation.description}</TittleWrapper>
          <RatingWrapper>
            <Rating
              value={0}
              max={5}
              itemIconStyle={RatingStyles.itemIconStyle}
              itemStyle={RatingStyles.itemStyle}
            />
          </RatingWrapper>
        </ContentWrapper>
        <WrapperCount>{counter}</WrapperCount>
      </Dialog>
    );
  }
}

export default graphql(CourseEvaluationtQuery, CourseEvaluationtQueryOptions)(CourseEvaluation);
