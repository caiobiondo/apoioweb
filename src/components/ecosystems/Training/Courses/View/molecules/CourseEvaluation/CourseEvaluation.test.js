import React from 'react';
import { shallow } from 'enzyme';
import { CourseEvaluation } from './CourseEvaluation';
import { FlatButton } from 'natura-ui';

const setup = propOverrides => {
  const intl = {
    formatMessage: value => `formatedMessage ${value}`,
  };

  const props = Object.assign(
    {
      user: {
        codigo: 1234,
      },
      courseId: 1,
      sellerId: 1,
      evaluations: [
        {
          id: 1,
          description: 'Description',
        },
        {
          id: 2,
          description: 'Description',
        },
      ],
      mutate: jest.fn().mockReturnValue(
        Promise.resolve({
          data: {
            addCourseEvaluations: {
              status: true,
              message: 'success',
            },
          },
        }),
      ),
      onLoadFinished: jest.fn(),
      intl,
    },
    propOverrides,
  );

  const result = shallow(<CourseEvaluation {...props} />);

  return {
    props,
    result,
  };
};

describe('Course Evaluation', () => {
  it('renders correctly when loading', () => {
    // given
    // when
    const { result } = setup({ fetchMore: jest.fn(), loading: true, evaluations: null });

    // then
    expect(result).toMatchSnapshot();
  });

  it('should notify onLoadFinished callback when not loading', () => {
    // given
    // when
    const { result, props } = setup({
      fetchMore: jest.fn(),
      loading: false,
    });
    result.instance().componentWillReceiveProps({ loading: false, evaluations: props.evaluations });

    // then
    expect(props.onLoadFinished).toBeCalledWith(false, false);
  });

  describe('Dialogs', () => {
    it('render evaluation dialog', () => {
      // given
      // when
      const { result, props } = setup({ fetchMore: jest.fn(), loading: false });

      // then
      expect(result).toMatchSnapshot();
    });

    it('renders feedback evaluation dialog', () => {
      // given
      // when
      const { result } = setup({ fetchMore: jest.fn(), loading: false });
      result.setState({ feedbackModalOpened: true });

      // then
      expect(result).toMatchSnapshot();
    });

    it('correctly call mutation', () => {
      // given
      // when
      const { result, props } = setup({ fetchMore: jest.fn(), loading: false });
      result.setState({ currentIndex: 1 });
      result.instance().handleClose();

      // then
      expect(props.mutate).toBeCalled();
    });
  });
});
