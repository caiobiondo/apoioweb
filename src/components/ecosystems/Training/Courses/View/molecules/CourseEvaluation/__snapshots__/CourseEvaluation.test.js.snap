// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Evaluation Dialogs render evaluation dialog 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={true}
        fullWidth={false}
        label={
            <FormattedMessage
                id="courseEvaluationNext"
                values={Object {}}
            />
        }
        labelPosition="after"
        labelStyle={
            Object {
                "fontFamily": "Roboto-Regular",
                "fontWeight": "700",
              }
        }
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={true}
        secondary={false}
    />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  key="evaluationModal"
  modal={false}
  onRequestClose={[Function]}
  open={true}
  repositionOnUpdate={true}
  title="Avaliação do curso"
  titleStyle={
    Object {
      "fontFamily": "NaturaSans-Bold",
      "fontSize": "25px",
    }
  }
>
  <CourseEvaluationstyles__ContentWrapper>
    <CourseEvaluationstyles__TitleWrapper>
      Description
    </CourseEvaluationstyles__TitleWrapper>
    <CourseEvaluationstyles__RatingWrapper>
      <Rating
        disabled={false}
        iconFilled={
          <ToggleStar
            color="#ff9800"
          />
        }
        iconHovered={
          <ToggleStarBorder
            color="#ff9800"
          />
        }
        iconNormal={
          <ToggleStarBorder
            color="#e0e0e0"
          />
        }
        itemIconStyle={
          Object {
            "height": "18",
            "width": "18",
          }
        }
        itemStyle={
          Object {
            "height": "18",
            "padding": "0",
            "width": "18",
          }
        }
        max={5}
        onChange={[Function]}
        readOnly={false}
        tooltipPosition="bottom-center"
        value={0}
      />
    </CourseEvaluationstyles__RatingWrapper>
  </CourseEvaluationstyles__ContentWrapper>
  <CourseEvaluationstyles__WrapperCount>
    1/2
  </CourseEvaluationstyles__WrapperCount>
</Dialog>
`;

exports[`Course Evaluation Dialogs renders feedback evaluation dialog 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={false}
        fullWidth={false}
        label={
            <FormattedMessage
                id="ok"
                values={Object {}}
            />
        }
        labelPosition="after"
        labelStyle={
            Object {
                "fontFamily": "Roboto-Regular",
                "fontWeight": "700",
              }
        }
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={true}
        secondary={false}
    />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  key="feedbackEvaluationModal"
  modal={false}
  open={true}
  repositionOnUpdate={true}
  title=""
  titleStyle={
    Object {
      "fontFamily": "NaturaSans-Bold",
      "fontSize": "25px",
    }
  }
/>
`;

exports[`Course Evaluation renders correctly when loading 1`] = `
<Loading
  background="transparent"
  visible={true}
  zIndex={9999}
/>
`;
